src/test/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImplTest.java

```diff
@@ -424,8 +425,21 @@ public void testBasicCredentials() throws Exception {
         filtered = scanner.filteredNodes(exec, new DescriptorMatchPredicate(EchoStep.DescriptorImpl.class));
         for (FlowNode f : filtered) {
             act = f.getPersistentAction(ArgumentsActionImpl.class);
-            Assert.assertEquals(ArgumentsAction.NotStoredReason.MASKED_VALUE, act.getArguments().get("message"));
-            Assert.assertNull(ArgumentsAction.getStepArgumentsAsString(f));
+            String id = f.getId();
+            switch (id) {
+                case "7":
+                    Assert.assertEquals("${PASSWORD}'", act.getArguments().get("message"));
+                    break;
+                case "8":
+                case "9":
+                    Assert.assertEquals("${USERNAME}", act.getArguments().get("message"));
+                    break;
+                case "16":
+                    Assert.assertEquals("${USERNAME} ${PASSWORD}", act.getArguments().get("message"));
+                    break;
+                default:
+                    Assert.fail();
+            }
```

Comment:
I would avoid hard-coding flow node IDs in tests. Maybe replace this with one simpler assertion that just makes sure the unmasked secrets aren't present:

```suggestion
            Assert.assertThat(act.getArguments().get("message"), allOf(
                not(containsString("bob")), 
                not(containsString("s3cr3t"))));
```

Reply:
resolved in 7c8022b5eec0e7ecad65341b3fac3b4c05fbae2b